doc/source/reference/c-api/array.rst

```diff
@@ -1555,8 +1555,296 @@ For all of these macros *arr* must be an instance of a (subclass of)
+   buffer, *size* is the 1D size of the array buffer, *stride* is the stride,
+   and *auxdata* is optional extra data for the loop.
+
+   The *traverse_context* is passed in because we may need to pass in
+   Interpreter state or similar in the future, but we don't want to pass in a
+   full context (with pointers to dtypes, method, caller which all make no sense
+   for a traverse function). We assume for now that this context can be just
+   passed through in the the future (for structured dtypes).
+
+.. c:type:: int (PyArrayMethod_GetTraverseLoop)( \
+                void *traverse_context, PyArray_Descr *descr, \
+                int aligned, npy_intp fixed_stride, \
+                PyArrayMethod_TraverseLoop **out_loop, NpyAuxData **out_auxdata, \
+                NPY_ARRAYMETHOD_FLAGS *flags)
+
+   Simplified get_loop function specific to dtype traversal
+
+   It should set the flags needed for the traversal loop and set *out_loop* to the
+   loop function, which must be a valid ``PyArrayMethod_TraverseLoop``
+   pointer. Currently this is used for zero-filling and clearing arrays storing
+   embedded references.
+
+.. c:type:: int (PyArrayMethod_PromoterFunction)(PyObject *ufunc, \
+                PyArray_DTypeMeta *op_dtypes[], PyArray_DTypeMeta *signature[], \
+                PyArray_DTypeMeta *new_op_dtypes[])
+
+   Type of the promoter function, which must be wrapped into a
+   ``PyCapsule`` with name ``"numpy._ufunc_promoter"``.
+
+   Note that currently the output dtypes are always ``NULL`` unless they are
+   also part of the signature. This is an implementation detail and could
+   change in the future. However, in general promoters should not have a
+   need for output dtypes.
+
+API Functions
+~~~~~~~~~~~~~
+
+These functions are part of the main numpy array API and were added along
+with the rest of the ArrayMethod API.
+
+.. c:function::  int PyUFunc_AddLoopFromSpec( \
+                         PyObject *ufunc, PyArrayMethod_Spec *spec)
+
+   Add loop directly to a ufunc from a given ArrayMethod spec.
+   the main ufunc registration function.  This adds a new implementation/loop
+   to a ufunc.  It replaces `PyUFunc_RegisterLoopForType`.
+
+.. c:function:: int PyUFunc_AddPromoter( \
+                        PyObject *ufunc, PyObject *DType_tuple, PyObject *promoter)
+
+   Register a new promoter for a ufunc. The first argument is the ufunc to
+   register the promoter with. The second argument is a Python tuple containing
+   DTypes or None matching the number of inputs and outputs for the ufuncs. The
+   last argument is a promoter is a function stored in a PyCapsule.  It is
+   passed the operation and requested DType signatures and can mutate it to
+   attempt a new search for a matching loop/promoter.
+
+.. c:function:: int PyUFunc_GiveFloatingpointErrors( \
+                        const char *name, int fpe_errors)
+
+    Signal a floating point error respecting the error signaling setting in
```

Comment:
Confused what "Signal a floating point error respecting..." means. Does this *set* a floating point error? Or does it signal that such floating point errors should be caught?

Reply:
reworded